session-replay/Sources/DatadogSessionReplay/Recorder/SnapshotBuilder/NodeRecorders/UITextFieldRecorder.swift

```diff
@@ -0,0 +1,106 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+
+internal struct UITextFieldRecorder: NodeRecorder {
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeSnapshotBuilder.Context) -> NodeSemantics? {
+        guard let textField = view as? UITextField else {
+            return nil
+        }
+
+        guard attributes.isVisible else {
+            return InvisibleElement.constant
+        }
+
+        // TODO: RUMM-2459
+        // Explore other (better) ways of infering text field appearance:
+
+        var editorProperties: UITextFieldWireframesBuilder.EditorFieldProperties? = nil
+        // Lookup the actual editor field's view in `textField` hierarchy to infer its appearance.
+        // Perhaps this can be do better by infering it from `UITextField` object in RUMM-2459:
+        dfsVisitSubviews(of: textField) { subview in
+            if subview.bounds == textField.bounds {
+                editorProperties = .init(
+                    backgroundColor: subview.backgroundColor?.cgColor,
+                    layerBorderColor: subview.layer.borderColor,
+                    layerBorderWidth: subview.layer.borderWidth,
+                    layerCornerRadius: subview.layer.cornerRadius
+                )
+            }
+        }
+
+        let text: String = {
+            guard let textFieldText = textField.text, !textFieldText.isEmpty else {
+                return textField.placeholder ?? ""
+            }
+            return textFieldText
+        }()
```

Comment:
/nit I kind of like the optional mapping thing, not a change request, up to you :)
```suggestion
        let text = textField.text.flatMap { $0.isEmpty ? textField.placeholder : $0 } ?? ""
```

Reply:
👌 Noted - but if it's not a blocker, I'll keep existing code 🙂. I'd be in favour of syntax sugar if there is more similar code around and wrapping it into simple & repetitive pattern would improve readability.